src/snowflake/connector/auth/sshagent.py

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
+
+import base64
+import hashlib
+import os
+import struct
+from datetime import datetime, timedelta, timezone
+from logging import getLogger
+from typing import Any
+
+import jwt
+import paramiko
+from jwt import algorithms as jwt_algorithms
+
+from ..errorcode import ER_CONNECTION_TIMEOUT, ER_INVALID_PRIVATE_KEY, ER_KEY_NAME_NOT_FOUND
+from ..errors import OperationalError, ProgrammingError
+from ..network import KEY_PAIR_AUTHENTICATOR
+from .by_plugin import AuthByPlugin, AuthType
+
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.backends import default_backend
+
+logger = getLogger(__name__)
+
+
+class RSASSHAlgorithm(jwt_algorithms.RSAAlgorithm):
+    def sign(self, msg: bytes, key_name: str) -> bytes:
+        agent = paramiko.Agent()
+        all_keys = agent.get_keys()
+        found_key = None
+        for key in all_keys:
+            if key.comment == key_name:
+                found_key = key
+                break
+
+        if found_key is None:
+            raise ProgrammingError(
+                msg=f"Failed to find key {key_name} in agent",
+                errno=ER_KEY_NAME_NOT_FOUND,
+            )
+
+        ssh_signature = found_key.sign_ssh_data(msg, "rsa-sha2-256")
+        pos = 0
+        length = struct.unpack(">I", ssh_signature[pos:pos + 4])[0]
+        pos += 4
+        pos += length
+        length = struct.unpack(">I", ssh_signature[pos:pos + 4])[0]
```

Comment:
length isn't used any more after this line.

Reply:
I'll fix this chunk.